solnlib/bulletin_rest_client.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright 2024 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+from solnlib import splunk_rest_client as rest_client
+from splunklib import binding
+from typing import Optional
+import json
+
+
+__all__ = ["BulletinRestClient"]
+
+
+class BulletinRestClient:
+    READ_ENDPOINT = "/services/admin/messages"
```

Comment:
I think the `admin` here points to the user. Would the API work if we have `/services/-/messages`?

Reply:
hmm I don't think this endpoint is user-dependent. instead `/services/admin/messages` we can just use`/services/messages` and it will create a message aswell. Endpoints like `/services/-/messages` or `/services/user/messages` etc. don't work. I went with `/services/admin/messages` just to be consistent with the documentation https://dev.splunk.com/enterprise/docs/developapps/manageknowledge/managebulletins/#Create-a-message